File: pingdart-java/src/main/java/com/pingdart/sdk/services/StorageService.java
package com.pingdart.sdk.services;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import okhttp3.*;

import java.io.File;
import java.io.IOException;

public class StorageService {
    private final OkHttpClient http;
    private final String baseUrl;
    private final Gson gson = new Gson();

    public StorageService(OkHttpClient http, String baseUrl) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    private String postRequest(String endpoint, JsonObject data) throws IOException {
        RequestBody body = RequestBody.create(gson.toJson(data), MediaType.get("application/json; charset=utf-8"));
        Request request = new Request.Builder()
                .url(baseUrl + "/files/" + endpoint)
                .post(body)
                .build();
        try (Response response = http.newCall(request).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String getStats() throws IOException {
        return postRequest("stats", new JsonObject());
    }

    public String listBuckets() throws IOException {
        return postRequest("get-buckets", new JsonObject());
    }

    public String createBucket(String name) throws IOException {
        JsonObject data = new JsonObject();
        data.addProperty("name", name);
        return postRequest("create-bucket", data);
    }

    public String deleteBucket(String bucketId) throws IOException {
        JsonObject data = new JsonObject();
        data.addProperty("id", bucketId);
        return postRequest("delete-bucket", data);
    }

    public String uploadFile(File file, String bucketName, String destination) throws IOException {
        MultipartBody.Builder builder = new MultipartBody.Builder()
                .setType(MultipartBody.FORM)
                .addFormDataPart("file", file.getName(),
                        RequestBody.create(file, MediaType.parse("application/octet-stream")));

        if (bucketName != null) builder.addFormDataPart("bucket_name", bucketName);
        if (destination != null) builder.addFormDataPart("destination", destination);

        Request request = new Request.Builder()
                .url(baseUrl + "/files/upload")
                .post(builder.build())
                .build();

        try (Response response = http.newCall(request).execute()) {
            if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
            return response.body().string();
        }
    }

    public String deleteFile(String filePath) throws IOException {
        JsonObject data = new JsonObject();
        data.addProperty("filePath", filePath);
        return postRequest("deleteFile", data);
    }
}

File: pingdart-php/src/Services/StorageService.php
<?php

namespace PingDart\SDK\Services;

use GuzzleHttp\Client;

class StorageService {
    private $http;

    public function __construct(Client $http) {
        $this->http = $http;
    }

    public function getStats() {
        try {
            $response = $this->http->post('files/stats');
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to get storage stats: " . $e->getMessage());
        }
    }

    public function listBuckets() {
        try {
            $response = $this->http->post('files/get-buckets');
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to list buckets: " . $e->getMessage());
        }
    }

    public function createBucket(string $name) {
        try {
            $response = $this->http->post('files/create-bucket', [
                'json' => ['name' => $name]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to create bucket: " . $e->getMessage());
        }
    }

    public function deleteBucket($id) {
        try {
            $response = $this->http->post('files/delete-bucket', [
                'json' => ['id' => $id]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to delete bucket: " . $e->getMessage());
        }
    }

    public function uploadFile(string $filePath, string $bucketName = null, string $destination = null) {
        try {
            $multipart = [
                [
                    'name'     => 'file',
                    'contents' => fopen($filePath, 'r'),
                    'filename' => basename($filePath)
                ]
            ];

            if ($bucketName) {
                $multipart[] = ['name' => 'bucket_name', 'contents' => $bucketName];
            }
            if ($destination) {
                $multipart[] = ['name' => 'destination', 'contents' => $destination];
            }

            $response = $this->http->post('files/upload', [
                'multipart' => $multipart
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to upload file: " . $e->getMessage());
        }
    }

    public function deleteFile(string $filePath) {
        try {
            $response = $this->http->post('files/deleteFile', [
                'json' => ['filePath' => $filePath]
            ]);
            return json_decode($response->getBody()->getContents(), true);
        } catch (\Exception $e) {
            throw new \Exception("Failed to delete file: " . $e->getMessage());
        }
    }
}
